connect/src/com/telenor/connect/ui/ConnectLoginButtonWithoutProgressBar.java

```diff
@@ -0,0 +1,169 @@
+package com.telenor.connect.ui;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.customtabs.CustomTabsCallback;
+import android.support.customtabs.CustomTabsClient;
+import android.support.customtabs.CustomTabsServiceConnection;
+import android.support.customtabs.CustomTabsSession;
+import android.text.TextUtils;
+import android.util.AttributeSet;
+import android.view.View;
+
+import com.telenor.connect.BrowserType;
+import com.telenor.connect.ConnectSdk;
+import com.telenor.connect.R;
+import com.telenor.connect.utils.ConnectUrlHelper;
+import com.telenor.connect.utils.CustomTabsHelper;
+
+import java.lang.ref.WeakReference;
+
+public class ConnectLoginButtonWithoutProgressBar extends ConnectWebViewLoginButton {
```

Comment:
Somewhat confusing that this class extends a class named _web view_ while at the same time containing fields named _custom tabs_. I assume it's common functionality? Can the common functionality be named something else? (Disclaimer: I don't know a whole lot about veb views vs custom tabs, so perhaps it makes sense to mix the two :P )

Reply:
So... On authentication, the button falls back on web view if custom tabs not supported? Perhaps documentation of the class could be helpful in explaining this? Doesn't have to be more than one sentence.